pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/ModelParser.groovy

```diff
@@ -139,16 +142,38 @@ class ModelParser implements Parser {
     public @CheckForNull ModelASTPipelineDef parse(ModuleNode src, boolean secondaryRun = false) {
         // first, quickly ascertain if this module should be parsed at all
         def pst = src.statementBlock.statements.find {
-            MethodCallExpression m = matchMethodCall(it)
-            return m != null && matchMethodName(m) == ModelStepLoader.STEP_NAME
+            return isDeclarativePipelineStep(it)
         }
 
-        if (pst==null) {
-            // Check if there's a 'pipeline' step somewhere nested within the other statements and error out if that's the case.
-            src.statementBlock.statements.each { checkForNestedPipelineStep(it) }
-            return null; // no 'pipeline', so this doesn't apply
+        if (pst != null) {
+            return parsePipelineStep(pst, secondaryRun)
+        } else {
+            // Look for the pipeline step inside methods named call.
```

Comment:
Oof.

Reply:
Actually a conscious decision on my part - we're only going to support `pipeline {...}` blocks in the `call()` method of `vars/*.groovy`. Limits the scope of transformation and the ways things could go weird.